src/services/permissions-checker.js

```diff
@@ -269,32 +147,8 @@ class PermissionsChecker {
     }
   }
 
-  async _retrievePermissions() {
-    return forestServerRequester
-      .perform('/liana/v3/permissions', this.environmentSecret, { renderingId: this.renderingId })
-      .then((responseBody) => {
-        PermissionsChecker.isrolesACLActivated = responseBody.meta
-          ? responseBody.meta.rolesACLActivated
-          : false;
-        this._setPermissions(responseBody.data);
-      })
-      .catch((error) => P.reject(new VError(error, 'Permissions error')));
-  }
-
-  _isPermissionExpired(permissionName) {
-    const currentTime = moment();
-    const lastRetrieve = PermissionsChecker.getLastRetrieveTime(this.renderingId, permissionName);
-
-    if (!lastRetrieve) {
-      return true;
-    }
-
-    const elapsedSeconds = currentTime.diff(lastRetrieve, 'seconds');
-    return elapsedSeconds >= PermissionsChecker.expirationInSeconds;
-  }
-
   async _retrievePermissionsAndCheckAllowed(collectionName, permissionName, permissionInfos) {
-    await this._retrievePermissions();
+    await PermissionsCache.retrievePermissions(this.environmentSecret, this.renderingId);
```

Comment:
Not the job of this service. The `PermissionsGetter` should decide by itself if it should retrieve the permissions or not

Reply:
Well actually no... because the permissions retrieval will depend on whether the cached permission is allowed or not!
So we'll need one service that will interfere with the other.
I left the retrievePermissions call in the permissions checker for now, let's discuss about it when possible.